packages/schemas/src/foundations/jsonb-types.ts

```diff
@@ -29,8 +29,32 @@ export const oidcModelInstancePayloadGuard = z
 
 export type OidcModelInstancePayload = z.infer<typeof oidcModelInstancePayloadGuard>;
 
+// Import from @logto/core-kit later, pending for new version publish
+export const webUriProtocolRegEx = /^https?:$/;
+export const mobileAppSchemeProtocolRegEx = /^[a-z][\d_a-z]*(\.[\d_a-z]+)+:$/;
+
+export const validateUrl = (urlString: string, type?: 'web' | 'mobile') => {
+  try {
+    const { protocol } = new URL(urlString);
+    const regex =
+      type === 'web'
+        ? webUriProtocolRegEx
+        : type === 'mobile'
+        ? mobileAppSchemeProtocolRegEx
+        : undefined;
```

Comment:
don't think we need this nested ternary. it would be fine to use web as a fallback?
```suggestion
    const regex =
      type === 'mobile'
        ? mobileAppSchemeProtocolRegEx
        : webUriProtocolRegEx;
```

Reply:
alright